extensions/sql-migration/src/api/azure.ts

```diff
@@ -259,6 +280,19 @@ export async function getAvailableSqlVMs(account: azdata.Account, subscription:
 	return response.response.data.value;
 }
 
+export async function getVMInstanceView(sqlVm: SqlVMServer, account: azdata.Account, subscription: Subscription): Promise<VirtualMachineInstanceView> {
+	const api = await getAzureCoreAPI();
+	const path = encodeURI(`/subscriptions/${subscription.id}/resourceGroups/${getResourceGroupFromId(sqlVm.id)}/providers/Microsoft.Compute/virtualMachines/${sqlVm.name}/instanceView?api-version=2022-08-01`);
+	const host = api.getProviderMetadataForAccount(account).settings.armResource?.endpoint;
+	const response = await api.makeAzureRestRequest(account, subscription, path, azurecore.HttpRequestMethod.GET, undefined, true, host);
+
+	if (response.errors.length > 0) {
```

Comment:
Would `errors` be `null` or `undefined`?

Reply:
shouldn't be possible in this case I believe, as the API will always contain an "errors" list, even if empty. Agreed the error handling could probably be improved here, but I reused the same implementation as all the other custom REST API requests we make here, and haven't seen any issues in the past